proxy/http/HttpTransact.cc

```diff
@@ -1971,11 +1971,11 @@ HttpTransact::OSDNSLookup(State *s)
       action = s->http_config_param->redirect_actions_self_action;
       TxnDebug("http_trans", "[OSDNSLookup] Self action - %d.", int(action));
     } else {
-      // Make sure the return value from contains is big enough for a void*.
-      intptr_t x{intptr_t(RedirectEnabled::Action::INVALID)};
       ink_release_assert(s->http_config_param->redirect_actions_map != nullptr);
-      ink_release_assert(s->http_config_param->redirect_actions_map->contains(s->dns_info.addr, reinterpret_cast<void **>(&x)));
-      action = static_cast<RedirectEnabled::Action>(x);
+      auto &addrs = *(s->http_config_param->redirect_actions_map);
+      auto spot   = addrs.find(swoc::IPAddr(&s->dns_info.addr.sa));
+      ink_release_assert(spot != addrs.end()); // Should always find an entry.
+      action = std::get<1>(*spot);
```

Comment:
Can we use a structured binding on find instead of std::get? Something like:

```cpp
auto &[key, action] = addrs.find...
```

Reply:
Maybe what you have is better with the explicit ink_release_assert.

Reply:
It's complicated. I initially used
```
auto && [ range, action ] = ...
```
but that shadowed an outer `action` and so didn't update appropriately.  I could have used
```
swoc::IPRange r;
std::tie(r, action) = ...
```
I decided the iterator style was reasonable primarily because at least it was the most similar to the original code. With relation to the assert, for ease of use reasons if the iterator is invalid
the range is bound to an invalid / empty range, which can be checked. Therefore it could be done like
```
auto && [ range, action ] = ...find...
ink_release_assert(!range.empty());
```
and that would do the same thing as checking for `== end()`.